File: event/src/test/java/de/danoeh/antennapod/event/playback/BufferUpdateEventTest.java
package de.danoeh.antennapod.event.playback;

import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public class BufferUpdateEventTest {

    @Test
    public void startedSetsHasStartedAndNotHasEnded() {
        BufferUpdateEvent event = BufferUpdateEvent.started();
        assertEquals(-1f, event.getProgress(), 0f);
        assertTrue(event.hasStarted());
        assertFalse(event.hasEnded());
    }

    @Test
    public void endedSetsHasEndedAndNotHasStarted() {
        BufferUpdateEvent event = BufferUpdateEvent.ended();
        assertEquals(-2f, event.getProgress(), 0f);
        assertFalse(event.hasStarted());
        assertTrue(event.hasEnded());
    }

    @Test
    public void progressUpdateStoresValueAndIsNeitherStartedNorEnded() {
        BufferUpdateEvent event = BufferUpdateEvent.progressUpdate(0.5f);
        assertEquals(0.5f, event.getProgress(), 0f);
        assertFalse(event.hasStarted());
        assertFalse(event.hasEnded());
    }

    @Test
    public void progressUpdateOfNegativeOneMasqueradesAsStarted() {
        BufferUpdateEvent event = BufferUpdateEvent.progressUpdate(-1f);
        assertTrue(event.hasStarted());
        assertFalse(event.hasEnded());
    }

    @Test
    public void progressUpdateOfNanIsNeitherStartedNorEnded() {
        BufferUpdateEvent event = BufferUpdateEvent.progressUpdate(Float.NaN);
        assertTrue(Float.isNaN(event.getProgress()));
        assertFalse(event.hasStarted());
        assertFalse(event.hasEnded());
    }

    @Test
    public void progressUpdateOfNegativeZeroIsNeitherStartedNorEnded() {
        BufferUpdateEvent event = BufferUpdateEvent.progressUpdate(-0.0f);
        assertEquals(-0.0f, event.getProgress(), 0f);
        assertFalse(event.hasStarted());
        assertFalse(event.hasEnded());
    }

    @Test
    public void progressUpdateOfPositiveZeroIsNeitherStartedNorEnded() {
        BufferUpdateEvent event = BufferUpdateEvent.progressUpdate(0.0f);
        assertEquals(0.0f, event.getProgress(), 0f);
        assertFalse(event.hasStarted());
        assertFalse(event.hasEnded());
    }
}

File: event/src/test/java/de/danoeh/antennapod/event/playback/BufferUpdateEventTest.kt
package de.danoeh.antennapod.event.playback

import java.lang.Float
import org.junit.Assert.assertEquals
import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test

class BufferUpdateEventTest {

    @Test
    fun startedSetsHasStartedAndNotHasEnded() {
        val event = BufferUpdateEvent.started()
        assertEquals(-1f, event.progress, 0f)
        assertTrue(event.hasStarted())
        assertFalse(event.hasEnded())
    }

    @Test
    fun endedSetsHasEndedAndNotHasStarted() {
        val event = BufferUpdateEvent.ended()
        assertEquals(-2f, event.progress, 0f)
        assertFalse(event.hasStarted())
        assertTrue(event.hasEnded())
    }

    @Test
    fun progressUpdateStoresValueAndIsNeitherStartedNorEnded() {
        val event = BufferUpdateEvent.progressUpdate(0.5f)
        assertEquals(0.5f, event.progress, 0f)
        assertFalse(event.hasStarted())
        assertFalse(event.hasEnded())
    }

    @Test
    fun progressUpdateOfNegativeOneMasqueradesAsStarted() {
        val event = BufferUpdateEvent.progressUpdate(-1f)
        assertTrue(event.hasStarted())
        assertFalse(event.hasEnded())
    }

    @Test
    fun progressUpdateOfNanIsNeitherStartedNorEnded() {
        val event = BufferUpdateEvent.progressUpdate(Float.NaN)
        assertTrue(Float.isNaN(event.progress))
        assertFalse(event.hasStarted())
        assertFalse(event.hasEnded())
    }

    @Test
    fun progressUpdateOfNegativeZeroIsNeitherStartedNorEnded() {
        val event = BufferUpdateEvent.progressUpdate(-0.0f)
        assertEquals(-0.0f, event.progress, 0f)
        assertFalse(event.hasStarted())
        assertFalse(event.hasEnded())
    }

    @Test
    fun progressUpdateOfPositiveZeroIsNeitherStartedNorEnded() {
        val event = BufferUpdateEvent.progressUpdate(0.0f)
        assertEquals(0.0f, event.progress, 0f)
        assertFalse(event.hasStarted())
        assertFalse(event.hasEnded())
    }
}
